Mathlib/Data/Polynomial/Derivation.lean

```diff
@@ -92,4 +92,67 @@ def mkDerivationEquiv : A ≃ₗ[R] Derivation R R[X] A :=
 @[simp] lemma mkDerivationEquiv_symm_apply (D : Derivation R R[X] A) :
     (mkDerivationEquiv R).symm D = D X := rfl
 
+end CommSemiring
 end Polynomial
+
+namespace Module
+
+open Polynomial
+/--
+Suppose `a` is an element of an `R`-algebra `A` and `M` is an `A`-module.
+Then `Module.comp_aeval R M a` is the `R[X]`-module with carrier `M`,
+where the action of `f : R[X]` is `f • m = (aeval a f) • m`.
+-/
+@[nolint unusedArguments] def comp_aeval (R M: Type*) {A : Type*} [CommSemiring R] [Semiring A]
+  [Algebra R A][AddCommMonoid M] [Module A M] [Module R M] [IsScalarTower R A M] (_ : A) := M
```

Comment:
Instead of using a type synonym, I think a one-field structure might be clearer here:
```
structure CompAEval (R M: Type*) {A : Type*} [CommSemiring R] [Semiring A]
    [Algebra R A][AddCommMonoid M] [Module A M] [Module R M] [IsScalarTower R A M] (_ : A) where
  val : M


variable {R A M} [CommSemiring R] [Semiring A] (a : A) [Algebra R A] [AddCommMonoid M] [Module A M]
  [Module R M] [IsScalarTower R A M]
def CompAEval.equiv : CompAEval R M a ≃ M where
  toFun := CompAEval.val
  invFun := CompAEval.mk
  left_inv := sorry
  right_inv := sorry

instance : AddCommMonoid <| CompAEval R M a     := (CompAEval.equiv a).addCommMonoid
instance : Module R <| CompAEval R M a          := (CompAEval.equiv a).module R
instance : Module A <| CompAEval R M a          := (CompAEval.equiv a).module A
```

Reply:
Actually, having made the change, I'm not sure it's better. What do you think?

Reply:
I'm happy with the change now.